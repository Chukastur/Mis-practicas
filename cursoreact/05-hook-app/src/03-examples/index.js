

export * from './MultipleCustomHooks'
export * from './LoadingMesage'
export * from './PokemonCard'
export * from '../hooks/useCounter'
export * from '../hooks/useFetch'
export * from '../hooks/useForm'

