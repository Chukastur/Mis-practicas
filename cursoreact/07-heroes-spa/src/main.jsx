import { StrictMode, react } from 'react';
import { createRoot } from 'react-dom/client';
import { BrowserRouter, RouterProvider } from "react-router-dom";

import {HeroesApp} from './HeroesApp';


createRoot(document.getElementById('root')).render(
  <StrictMode>
    
   <BrowserRouter>
    <HeroesApp />
  </BrowserRouter>
   
  </StrictMode>
);
