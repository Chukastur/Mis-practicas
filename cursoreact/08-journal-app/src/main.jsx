
import { BrowserRouter } from 'react-router-dom';
import { createRoot } from 'react-dom/client'
import { Provider } from 'react-redux'
import { StrictMode } from 'react'

import { store } from './store'
import { JournalApp } from './JournalApp'
import './styles.css'

createRoot(document.getElementById('root')).render(

  <StrictMode>
    <Provider store={store}>
      <BrowserRouter>
        <JournalApp />
      </BrowserRouter>
    </Provider>
  </StrictMode>

);
