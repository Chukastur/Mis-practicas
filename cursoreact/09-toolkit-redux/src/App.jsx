import { useDispatch, useSelector } from 'react-redux';
import { decrement, increment, incrementBy } from './store';
import './App.css';

function App() {

  const { counter } = useSelector ( state => state.counter);
  const dispatch = useDispatch();

  return (
    <>
      <h1>Vite + React</h1>
      <p>count is { counter }</p>
      <div>
        <button onClick={ () => dispatch ( increment () ) }>Suma</button>
          
        <button onClick={ () => dispatch ( decrement () ) }>Resta</button>

        <button onClick={ () => dispatch ( incrementBy (2) ) }>Suma 2</button> 
      </div>
    </>

  );
};

export default App;
