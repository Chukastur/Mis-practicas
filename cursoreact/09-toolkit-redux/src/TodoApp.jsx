import { useState } from "react";
import { useGetTodoByIdQuery, useGetTodosQuery } from "./store/apis/todosApi";


export const TodoApp = () => {

  const [todoId, setTodoId ] = useState (1);
                                                 //destructuramos y renombramos data
  //const { data: todos = [] , isLoading } = useGetTodosQuery ();
  const { data: todo, isLoading } = useGetTodoByIdQuery (todoId); 
  
  const nextTodo = () => { 
    setTodoId ( todoId + 1 );
  };

  const prevTodo = () => { 
    if ( todoId === 1) return;
    setTodoId ( todoId - 1 );
  };
  

  return (
   <>
      <h1>Todos - RTK Query</h1>
      <hr />
      <h4>isLoading...{ isLoading ? 'Tue' : 'False'}</h4>

      <pre>{ JSON.stringify ( todo )}</pre>

      <button onClick={ prevTodo }>
        Prev Todo
      </button>
      <button onClick={ nextTodo }>
        Next Todo
      </button>
   </>
  );
};
