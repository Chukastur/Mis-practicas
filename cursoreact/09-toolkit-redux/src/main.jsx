import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { Provider } from 'react-redux';
import { store } from './';
// import App from './App.jsx';
import { PokemonApp } from './PokemonApp.jsx';
import './index.css';
import { TodoApp } from './TodoApp.jsx';



createRoot(document.getElementById('root')).render(
  <StrictMode>

    <Provider store={ store }>
      
      {/* <App />       */}
      {/* <PokemonApp /> */}
      {<TodoApp />}

    </Provider>

  </StrictMode>
);
  