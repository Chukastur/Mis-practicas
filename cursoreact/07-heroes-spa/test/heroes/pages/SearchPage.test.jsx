import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter, useNavigate } from "react-router-dom";
import { SearchPage } from "../../../src/heroes";

const mockedUseNavigate = jest.fn();

jest.mock ("react-router-dom", () => ({
    ...jest.requireActual ( "react-router-dom"),
    useNavigate : () => mockedUseNavigate
}));

describe('Pruebas en <SearchPage />', () => { 

    beforeEach ( () => jest.clearAllMocks() );


    test('debe de mostrarse con los valores por defecto', () => { 

        const { container } = render (
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        );

        expect ( container ).toMatchSnapshot ();

    });

    test('debwe de mostrar a Batman y el input con el valor del query String', () => { 

        render (
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole ( 'textbox' );
        expect ( input.value ).toBe ( 'batman' )

        const img = screen.getByRole ( 'img' );
        expect ( img.src ).toContain ( 'batman.jpg' );

        const divSearch = screen.getByLabelText ( 'testItem' );
        expect ( divSearch.style.display ).toBe ( 'none' );
    });

    test('debe de mostrer un error si no se encuentra el hero (batman123) ', () => { 

        render (
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        );

        const divError = screen.getByLabelText ( 'testItem2' );
        expect ( divError.style.display ).toBe ( '' );

    });

    test('debe de llamar el navigate a la pantalla nueva', () => {  

        const inputValue = 'superman';

        render (
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
        );
        
        const input = screen.getByRole ( 'textbox' );
        fireEvent.change ( input, { target: { name: 'searchText', value: inputValue }} );
        const form = screen.getByLabelText ( 'form' );
        fireEvent.submit ( form );

        expect ( mockedUseNavigate ).toHaveBeenCalledWith ( `?q=${ inputValue }` );
        

    });

}); 