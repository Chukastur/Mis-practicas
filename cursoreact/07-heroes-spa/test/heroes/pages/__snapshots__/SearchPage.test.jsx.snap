// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <SearchPage /> debe de mostrarse con los valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Búsqueda
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Busca un héroe"
          type="text"
          value=""
        />
        <button
          class="btn btn-dark mt-2"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        aria-label="testItem"
        class="alert alert-dark animate__animated animate__fadeIn"
      >
        Buscar un héroe
      </div>
      <div
        aria-label="testItem2"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hay resultados con 
        <b />
        ...
      </div>
    </div>
  </div>
</div>
`;
